### PR TITLE
add si7021 module

### PR DESCRIPTION
**Pull request by fetchbot**
- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

Adds a module for the Si7021 humidity and temperature sensor.

### CODE REVIEW

**Comment by FrankX0** [comment]
Works great!

**Comment by devsaurus** [comment]
Thanks again @fetchbot!

For the records, this addition obsoleted the Lua module implementation. See https://github.com/nodemcu/nodemcu-firmware/commit/f1bc858021774f8ce189d2fdc5b103b2712ea021.

File: mkdocs.yml [hunk 1]
```diff
@@ -71,6 +71,7 @@ pages:
         - 'rtcfifo': 'en/modules/rtcfifo.md'
         - 'rtcmem': 'en/modules/rtcmem.md'
         - 'rtctime': 'en/modules/rtctime.md'
+        - 'si7021' : 'en/modules/si7021'
```
Comment (marcelstoer) [inline mkdocs.yml]: Here a `.md` is missing. Our RTD builds failed and thus https://nodemcu.readthedocs.io/en/latest/en/modules/si7021 return HTTP 404. I'll fix this.

File: docs/en/modules/si7021.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# Si7021 Module
```
Comment (marcelstoer) [inline docs/en/modules/si7021.md]: The functions should be ordered alphabetically. I'll fix that as well.

**Pull request merged**